### PR TITLE
tutorials/09_use_custom_engine: Add actual build&run instructions

### PR DESCRIPTION
**Pull request by peci1**
# 🦟 Bug fix

Fixes https://github.com/gazebosim/gazebo_test_cases/issues/1428

## Summary
The tutorial included wrong instructions for building the example. This PR fixes that.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: tutorials/09_use_custom_engine.md [hunk 1]
```diff
@@ -158,13 +158,56 @@ While we won't go into detail, here is an example to test our new
...
-- TPE: [EntityManagementFeatures.hh](https://github.com/gazebosim/gz-physics/blob/ign-physics2/tpe/plugin/src/EntityManagementFeatures.hh) and [EntityManagementFeatures.cc](https://github.com/gazebosim/gz-physics/blob/ign-physics2/tpe/plugin/src/EntityManagementFeatures.cc)
+- Dartsim: [EntityManagementFeatures.hh](https://github.com/gazebosim/gz-physics/blob/main/dartsim/src/EntityManagementFeatures.hh) and [EntityManagementFeatures.cc](https://github.com/gazebosim/gz-physics/blob/main/dartsim/src/EntityManagementFeatures.cc)
+- TPE: [EntityManagementFeatures.hh](https://github.com/gazebosim/gz-physics/blob/main/tpe/plugin/src/EntityManagementFeatures.hh) and [EntityManagementFeatures.cc](https://github.com/gazebosim/gz-physics/blob/main/tpe/plugin/src/EntityManagementFeatures.cc)
+
+## Compile and run the example
+
+Clone the source code, create a build directory and use `cmake` to compile the code:
+
+```{.sh}
```
Comment (ahcorde) [inline tutorials/09_use_custom_engine.md:168]:
```suggestion
```bash
```

Comment (ahcorde) [reply tutorials/09_use_custom_engine.md:168]: Same with the others

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: tutorials/09_use_custom_engine.md (hunk 1 above)
Comment (peci1) [reply tutorials/09_use_custom_engine.md]: Updated

**Review by ahcorde — APPROVED** [review]

**Review by iche033 — APPROVED** [review]

**Comment by azeey** [comment]
Looks like there are some CI issues, but I'll go ahead and merge this since it's only updating documentation.

**Comment by scpeters** [comment]
https://github.com/Mergifyio backport gz-physics7

**Pull request merged**